website/blog/2025-09-29-blog-post-title.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Bringing NumPy's type-completeness score to nearly 90%
+description: We tell the story of how we brought NumPy's type-completeness score from ~33% to nearly 90%
+slug: numpy-type-completeness
+authors: [marcogorelli]
+tags: [typechecking, news]
+hide_table_of_contents: false
+---
+
+[NumPy](https://numpy.org/) is one of the most downloaded packages in the Python ecosystem. Therefore, any incremental improvement can have a large impact on the data science ecosystem. In particular, improvements related to static typing can improve developer experience and help downstream libraries write safer code. We'll tell the story about how we helped bring its type-completeness score to nearly 90% from an initial 33%.
```

Comment:
Because [NumPy](https://numpy.org/) is one of the most downloaded packages in the Python ecosystem, even incremental improvement scan have a large impact on the data science ecosystem. In particular, static typing improvements improve developer experience and help downstream libraries write safer code. [...]